iRecruit-ui/src/components/Main/Main.jsx

```diff
@@ -109,6 +114,27 @@ function Main() {
 
   };
 
+  // Function to handle clicking on "Start Application" button
+  const handleStartApplication = (jobId) => {
+    setSelectedPostId(jobId);
+    setFlippedPostId(jobId);
+
+    const preferredPost = posts.find((post) => post.slug === jobId);
+    if (preferredPost) {
+      setPreferredPosts((prevPreferredPosts) => [...prevPreferredPosts, preferredPost]);
+    }
+  };
+
+  // Function to handle clicking on "Like" button
+  const handleLike = (job) => {
+    setLikedPosts((prevLikedPosts) => [...prevLikedPosts, job]);
+  };
+
+  // Function to handle clicking on "Dislike" button
+  const handleDislike = (job) => {
+    setDislikedPosts((prevDislikedPosts) => [...prevDislikedPosts, job]);
+  };
```

Comment:
Look up the `useCallback` hook if you want some more advanced React topics :)

Reply:
Okay, sure! Thank you